### PR TITLE
Add branch 18.0 [WAIT FOR MERGE]

### PR DESCRIPTION
**Pull request by simahawk**
Addresses v18 checklist https://github.com/OCA/maintainer-tools/issues/628

How to obtain this result? Easy w/ https://github.com/OCA/repo-maintainer/pull/20

```
oca-repo-add-branch --conf-dir ./conf/ --branch 18.0
```
That's it.

**WAIT FOR MERGE TILL REST OF THE TOOLS ARE READY**

### CODE REVIEW

**Comment by simahawk** [comment]
I still have to address the use case to reuse answers from previous branch https://github.com/OCA/maintainer-tools/pull/541

File: conf/repo/maintainer.yml [hunk 1]
```diff
@@ -1,14 +1,16 @@
 maintainer-quality-tools:
-  branches: []
+  branches:
+    - "18.0"
   category: Maintainer
-  default_branch: master
+  default_branch: "18.0"
```
Comment (sbidoul) [inline conf/repo/maintainer.yml:5]: Careful with these

**Comment by etobella** [comment]
@simahawk @sbidoul  Which answers do you want to reuse? Name description and so on are already on the configuration :thinking:

**Comment by pedrobaeza** [comment]
And this should wait until we update the configurations for this new branch.

File: conf/repo/maintainer.yml (hunk 1 above)
Comment (simahawk) [reply conf/repo/maintainer.yml:5]: Yep, I've seen it. I'll handle some blacklisting :)

**Comment by simahawk** [comment]
> @simahawk @sbidoul Which answers do you want to reuse? Name description and so on are already on the configuration 🤔

Agree, see https://github.com/OCA/maintainer-tools/pull/541#issuecomment-2348122119

**Comment by sbidoul** [comment]
@simahawk can you explain how you this this PR in the PR description, as reference for the future?

File: conf/repo/maintainer.yml (hunk 1 above)
Comment (simahawk) [reply conf/repo/maintainer.yml]: Fixed

**Comment by simahawk** [comment]
> @simahawk can you explain how you this this PR in the PR description, as reference for the future?

Sure, I'll also improve the README of the tool.

File: conf/repo/maintainer.yml [hunk 2]
```diff
@@ -1,12 +1,14 @@
 maintainer-quality-tools:
-  branches: []
+  branches:
+    - "18.0"
```
Comment (sbidoul) [inline conf/repo/maintainer.yml:3]: BTW, maintainer-quality-tools is archived, so we can remove it from here

File: conf/repo/ansible.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ansible-odoo:
-  branches: []
+  branches:
+    - "18.0"
```
Comment (simahawk) [inline conf/repo/ansible.yml:3]: wrong

File: conf/repo/connector.yml [hunk 4]
```diff
@@ -161,13 +170,15 @@ connector-magento:
...
   category: Connector
-  default_branch: "11.0"
+  default_branch: "18.0"
   name: Connect odoo with magento
   psc: connector-magento-maintainers
   psc_rep: connector-magento-maintainers-psc-representative
 connector-magento-php-extension:
-  branches: []
+  branches:
+    - "18.0"
```
Comment (simahawk) [inline conf/repo/connector.yml:181]: wrong

File: conf/repo/maintainer.yml [hunk 5]
```diff
@@ -1,12 +1,14 @@
...
   category: Maintainer
   default_branch: master
   name: Qa tools for odoo maintainers
   psc: community-maintainers
   psc_rep: community-maintainers-psc-representative
 maintainer-tools:
-  branches: []
+  branches:
```
Comment (simahawk) [inline conf/repo/maintainer.yml:10]: wrong

File: conf/repo/mirrors.yml [hunk 6]
```diff
@@ -1,5 +1,6 @@
 mirrors-flake8:
-  branches: []
+  branches:
+    - "18.0"
```
Comment (simahawk) [inline conf/repo/mirrors.yml:3]: wrong

File: conf/repo/oca.yml [hunk 7]
```diff
@@ -1,12 +1,14 @@
 oca-addons-repo-template:
-  branches: []
+  branches:
+    - "18.0"
```
Comment (simahawk) [inline conf/repo/oca.yml:3]: wrong

File: conf/repo/oca.yml [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
   category: Oca
   default_branch: master
   name: Oca repository template
   psc: community-maintainers
   psc_rep: community-maintainers-psc-representative
 oca-ci:
-  branches: []
+  branches:
```
Comment (simahawk) [inline conf/repo/oca.yml:10]: wrong

File: conf/repo/maintainer.yml (hunk 2 above)
Comment (simahawk) [reply conf/repo/maintainer.yml:3]: I'll do a separate PR for this.

File: conf/repo/oca.yml (hunk 8 above)
Comment (simahawk) [reply conf/repo/oca.yml]: Should be all fine now

File: conf/repo/maintainer.yml (hunk 2 above)
Comment (simahawk) [reply conf/repo/maintainer.yml]: [Shall we archive https://github.com/OCA/runbot-addons too?](https://github.com/OCA/repo-maintainer-conf/pull/37)

File: conf/repo/oca.yml [hunk 9]
```diff
@@ -20,8 +20,9 @@ oca-custom:
     - "12.0"
     - "13.0"
     - "14.0"
+    - "18.0"
   category: Oca
-  default_branch: "14.0"
```
Comment (etobella) [inline conf/repo/oca.yml:24]: I think this file shouldn't be modified

File: conf/repo/odoo-community.yml [hunk 10]
```diff
@@ -21,15 +21,17 @@ odoo-pim:
...
+  default_branch: "18.0"
   name: Odoo Pim
   psc: product-maintainers
   psc_rep: product-maintainers-psc-representative
 odoo-pre-commit-hooks:
-  branches: []
+  branches:
+    - "18.0"
   category: Odoo
-  default_branch: main
```
Comment (etobella) [inline conf/repo/odoo-community.yml:32]: I think this shouldn't be modified

File: conf/repo/odoo-community.yml [hunk 11]
```diff
@@ -52,8 +54,9 @@ pylint-odoo:
   branches:
     - "7.0"
     - "8.0"
+    - "18.0"
   category: Pylint
-  default_branch: main
```
Comment (etobella) [inline conf/repo/odoo-community.yml:56]: I think this shouldn't be modified

File: conf/repo/oca.yml (hunk 9 above)
Comment (simahawk) [reply conf/repo/oca.yml:24]: fixed all the "main" branch cases (in the tool)

Comment (simahawk) [reply conf/repo/oca.yml]: should be fine now thanks to the key `manual_branch_mgmt: true`

**Comment by pedrobaeza** [comment]
@simahawk I have already switched default branch of OCB to 18.0 FYI in a direct commit for continuing with the rest of the needed changes for the foundation.

**Comment by sbidoul** [comment]
I suggest making a PR that creates the 18 branch on a couple of repos first, where someone is ready to do a PR to test the repo template.

**Comment by simahawk** [comment]
Rebased

**Comment by simahawk** [comment]
FTR v18 branch tested on server-env

- https://github.com/OCA/repo-maintainer-conf/pull/40
- https://github.com/OCA/server-env/pull/204

**Review by pedrobaeza — COMMENTED** [review]
Do we want to switch the default branch for all the repos to version 18?

**Comment by sbidoul** [comment]
> Do we want to switch the default branch for all the repos to version 18?

I'm not sure. It would give the impression of empty repos everywhere in OCA.

**Comment by simahawk** [comment]
> Do we want to switch the default branch for all the repos to version 18?

It was in the checklist :wink: 

> I'm not sure. It would give the impression of empty repos everywhere in OCA.

Yes, same feeling here. Yet, this is true even when you have 1 single module migrated :)

Personally I don't mind. I can roll back this change. What do you prefer?

We should probably improve the readme of the repos by including a message and a link to pull requests.
Something like "Migration in progress, bla bla bla. Please check $link".

**Comment by simahawk** [comment]
@pedrobaeza @sbidoul there you go https://github.com/OCA/repo-maintainer-conf/pull/41

**Pull request closed without merging**